addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -977,9 +977,21 @@ const Wysiwyg = Widget.extend({
                         name: el.dataset.fileName || '',
                         data: el.getAttribute('src').split(',')[1],
                         is_image: true,
+                        ...(this.options.imageAsPublicAttachment ? {} : this.options.recordInfo),
```

Comment:
Now that I think about it... did you look at what is inside `recordInfo` in the case of website ? I think it is likely to be `ir.ui.view` and therefore end up as public anyway, without requiring an explicit option.